sql/updates/latest-dev.sql

```diff
@@ -325,3 +325,72 @@ ALTER TABLE _timescaledb_catalog.tablespace
     ADD CONSTRAINT tablespace_hypertable_id_fkey FOREIGN KEY (hypertable_id) REFERENCES _timescaledb_catalog.hypertable(id) ON DELETE CASCADE;
 
 CREATE SCHEMA _timescaledb_debug;
+
+-- Migrate existing compressed hypertables to new internal format
+DO $$
+DECLARE
+  chunk regclass;
+  hypertable regclass;
+  ht_id integer;
+  index regclass;
+  column_name name;
+  cmd text;
+BEGIN
+  SET timescaledb.restoring TO ON;
+
+  -- Detach compressed chunks from their parent hypertables
```

Comment:
Since NO INHERIT does not rewrite the table it should finish instantly but it does indeed require AccessExclusive lock. But AccessExclusive locks are required by most non-patch releases I think this only becomes a problem when we try to do expensive data migrations like rewriting tables which isn't the case here.